### PR TITLE
11 Local Database

### PR DESCRIPTION
**Pull request by craftamap**
This pull request also contains #25 , but resolves all merge issues.

Please take a look at here asap.

### CODE REVIEW

File: app/src/main/java/de/hsaugsburg/teamulster/sohappy/analyzer/ImageAnalyzer.kt [hunk 1]
```diff
@@ -18,10 +23,46 @@ import kotlin.concurrent.thread
...
+                else -> imageAnalyzerState
+            }
+        }
+        imageAnalyzerState = when (stateMachine.getCurrentMachineState()) {
+            is Start -> ImageAnalyzerState.NONE
+            is WaitingForFace, is TakeABreath -> ImageAnalyzerState.FACE_DETECTION
+            is WaitingForSmile -> ImageAnalyzerState.SMILE_DETECTION
+            is Questions, is NoSmile -> ImageAnalyzerState.CANCEL
+            else -> imageAnalyzerState
+        }
```
Comment (AnnaGsell) [inline app/src/main/java/de/hsaugsburg/teamulster/sohappy/analyzer/ImageAnalyzer.kt:62]: is this twice the same doing? 'Cause I can't see the difference

Comment (craftamap) [reply app/src/main/java/de/hsaugsburg/teamulster/sohappy/analyzer/ImageAnalyzer.kt:62]: Once its on start, onee its in the listener.

**Review by diehmlukas — APPROVED** [review]

**Review by mac641 — APPROVED** [review]
Looks good to me 👍🏿  
Request changes in previous PR have been applied

**Review by diehmlukas — DISMISSED** [review]

**Review by AnnaGsell — APPROVED** [review]

**Review by diehmlukas — APPROVED** [review]

**Review by mac641 — APPROVED** [review]
🚁 117

**Pull request merged**